app/modules/site_settings/resources.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=bad-continuation
+"""
+RESTful API Site Settings resources
+--------------------------
+"""
+
+import logging
+from pathlib import Path
+
+from flask import redirect, url_for
+from flask_restx_patched import Resource
+from flask_restx._http import HTTPStatus
+
+from app.extensions import db
+from app.extensions.api import abort, Namespace
+from app.extensions.api.parameters import PaginationParameters
+from app.modules.fileuploads.models import FileUpload
+from app.modules.users import permissions
+from app.modules.users.permissions.types import AccessOperation
+
+from . import schemas, parameters
+from .models import SiteSetting
+
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+api = Namespace(
+    'site-settings', description='Site Settings'
+)  # pylint: disable=invalid-name
+
+
+@api.route('/')
+@api.login_required(oauth_scopes=['site-settings:read'])
+class SiteSettings(Resource):
+    """
+    Manipulations with Site Settings.
+    """
+
+    @api.permission_required(permissions.AdminRolePermission())
+    @api.parameters(PaginationParameters())
+    @api.response(schemas.BaseSiteSettingSchema(many=True))
+    def get(self, args):
+        """
+        List of SiteSetting.
+
+        Returns a list of SiteSetting starting from ``offset`` limited by ``limit``
+        parameter.
+        """
+        return (
+            SiteSetting.query.order_by('key').offset(args['offset']).limit(args['limit'])
+        )
+
+    @api.permission_required(permissions.AdminRolePermission())
```

Comment:
Should use ModuleAccessPermission for control of what users can do what operations on a module